src/build/build.ts

```diff
@@ -88,66 +83,45 @@ export async function build(
           .pipe(gulpif(/\.html$/, new HTMLOptimizeStream(optimizeOptions.html)))
           .pipe(polymerProject.rejoinHtml());
 
-  let buildStream = mergeStream(sourcesStream, depsStream);
-
-  let unbundledPhase = forkStream(buildStream)
-                           .once(
-                               'data',
-                               () => {
-                                 logger.info('Generating build/unbundled...');
-                               })
-                           .pipe(gulpif(
-                               options.insertDependencyLinks || false,
-                               new PrefetchTransform(polymerProject)))
-                           .pipe(dest(unbundledBuildDirectory));
-
-  let bundledPhase = forkStream(buildStream)
-                         .once(
-                             'data',
-                             () => {
-                               logger.info('Generating build/bundled...');
-                             })
-                         .pipe(polymerProject.bundler)
-                         .pipe(dest(bundledBuildDirectory));
+  let buildStream =
+      <NodeJS.ReadableStream>mergeStream(sourcesStream, depsStream)
+          .once('data', () => {
+            logger.debug('Analyzing build dependencies...');
+          });
 
-  let swPrecacheConfig = path.resolve(
-      config.root, options.swPrecacheConfig || 'sw-precache-config.js');
-  let loadSWConfig = parsePreCacheConfig(swPrecacheConfig);
-
-  loadSWConfig.then((swConfig) => {
-    if (swConfig) {
-      logger.debug(`Service worker config found`, swConfig);
-    } else {
-      logger.debug(`No service worker configuration found at ${swPrecacheConfig
-                   }, continuing with defaults`);
-    }
+  if (options.bundle) {
+    buildStream = buildStream.pipe(polymerProject.bundler);
+  }
+
+  if (options.insertPrefetchLinks) {
+    buildStream = buildStream.pipe(new PrefetchTransform(polymerProject));
+  }
+
+  buildStream = buildStream
+                    .once(
+                        'data',
+                        () => {
+                          logger.info('Generating build/ directory...');
+                        })
+                    .pipe(dest(buildDirectory));
+
+  // While the build is in progress, parse the sw precache config
+  const swConfig = await parsePreCacheConfig(swPrecacheConfig);
+  if (swConfig) {
+    logger.debug(`Service worker config found`, swConfig);
+  } else {
+    logger.debug(`No service worker configuration found at ${swPrecacheConfig
+                  }, continuing with defaults`);
+  }
+
+  // Once the build stream is complete, create a service worker for the build
+  await waitFor(buildStream);
+  await addServiceWorker({
+    buildRoot: buildDirectory,
+    project: polymerProject,
+    swPrecacheConfig: swConfig || undefined,
```

Comment:
`swConfig || undefined` is a little awkward, can it be fixed in `addServiceWorkers`'s or `swConfig`'s declarations?